backend/src/main/java/org/eclipse/tractusx/puris/backend/common/edc/logic/service/EdcAdapterService.java

```diff
@@ -834,7 +830,7 @@ private JsonNode getAasSubmodelDescriptors(String manufacturerPartId, String man
                             var aasJson = objectMapper.readTree(body2String);
                             var submodelDescriptors = aasJson.get("submodelDescriptors");
                             if (submodelDescriptors != null) {
-                                failed = false;
+                                criticalFailure = false;
```

Comment:
Please rename to edcFailure. I would only like to invalidate the Contract Data, if there is an EDC issue. Same as here an empty DTR result is fine.

Beside that I tried with Delivery Information. There we create multiple contracts. It might be an issue that we receive an error as in the test setup no delivery is available and an error is created during mapping (but it's actually caught). 

I think the error comes from proxyPull call in line 581 which is not successful.

Reply:
Regarding the requested renaming: I think it is not quite accurate to label failures in this category generally as "edcFailures". Other causes are just as possible. Example: The other party is running a version of DTR which has a different API that sends responses with an unexpected syntax, etc.

Reply:
That's a point, but that would be misconfiguration on provider side. 

The AAS spec standardizes routes, inputs and return values for a specific version (we use 3.0), e.g. [this file](https://github.com/admin-shell-io/aas-specs-api/blob/main/AssetAdministrationShellRegistryServiceSpecification/V3.0_SSP-001.yaml) specifies the registry interface of the DTR. 

What could happen from my point here:
- Misconfiguration of EDC assets (e.g. recent update of the CX DTR changed the base path, registered under wrong version)
- Services unavailable

Actually not sure what would be the correct approach in that scenario: if I try to recontract and get a contract for the same asset, I win nothing, but maybe the old offer has been invalidated (which wouldn't make sense as the asset is also wrong?) so that I found a new one. I guess full error handling should be something to investigate further.

Maybe criticalFailure is OK as naming, but we then should likely not even retry contracting more than once given that it's a configuration issue. What do you think?

Reply:
Well it's quite clear that dropping a contract, when a criticalFailure was detected is in way a "Silver Bullet" in every situation. But at least it's an approach that sometimes can help, the famous quote "Have you tried turining it off and on again?" comes to mind .... ;)
And at the very least, this approach makes sure, that the backend may eventually recover, if the other partner has fixed the assumed issue on his side and maybe did a restart of his system.

Reply:
As we already discussed yesterday, the failing within the DeliveryInformation was caused by the fact that our SammMapper-Classes were unable to create an empty response, when there is not a single entity of the requested type in the database. 
I have now enabled the various SammMappers to return empty responses.